beta-src/src/components/ui/WDSaveReadyButtons.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+
+import { Stack } from "@mui/material";
+import WDButton from "./WDButton";
+import SaveStatus from "../../types/saveStatus";
+
+interface saveReadyProps {
+  gameState: SaveStatus;
+  setGameState: React.Dispatch<React.SetStateAction<SaveStatus>>;
+}
+const WDSaveReadyButtons: React.FC<saveReadyProps> = function ({
+  gameState,
+  setGameState,
+}) {
+  const { save, ready } = gameState;
+
+  return (
+    <Stack direction="row" spacing={2} alignItems="center">
+      <WDButton
+        color="primary"
+        disabled={ready}
+        onClick={() => {
+          setGameState((preState) => ({
```

Comment:
the parent can manage the state....so just pass gameState and onClick and then onClick is just:

onClick={onClick}